sdks/python/apache_beam/ml/inference/api.py

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Tuple, TypeVar, Union
+
+import apache_beam as beam
+
+
+class PyTorchDevice(Enum):
+  CPU = 1
+  GPU = 2
+
+
+class SklearnSerializationType(Enum):
+  PICKLE = 1
+  JOBLIB = 2
+
+
+@dataclass
+class BaseModelSpec:
+  model_uri: str
```

Comment:
I've been thinking about this.  What if the model is hosted on some service that requires a more complex loading strategy, like gcs.

https://stackoverflow.com/questions/45920895/read-a-file-from-gcs-in-apache-beam

Will the model_uri be adequate for us?

Reply:
Could we, as future feature requests, add a wrappers to support these different services? 
e.g.
```
class RemoteServiceType(enum):
    beam_file_systems = 1
    gcs_client = 2  
```

Reply:
Is there a need for any prescription for specific data in the baseclass? The key property of the base model spec is the methods it provides (to instantiate a runnable model) but it shouldn't dictate how that model is defined/referenced.

Reply:
I'm not following-- could you clarify? Also, we've scheduled a meeting to sync on this, so perhaps it we can get alignment then.

Reply:
Resolving per our discussion: Only subclasses only need to look at `model_uri`, so we should put it there.